### PR TITLE
Test/poll daemon

### PR DESCRIPTION
**Pull request by cryptix**
In the quest for more stable tests (#844) we also want a better way to notice when the daemon is ready to accept http requests.

Previously we had a sleep/grep construction. This uses a small go program that polls an endpoint and returns with exit code zero when it returns `http 200 ok`

### CODE REVIEW

**Comment by cryptix** [comment]
I forgot to mention that this also adds the `/version` endpoint to the JSON API on port 5001.

File: cmd/pollEndpoint/main.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	flag.Parse()
+
+	// extract address from host flag
+	addr, err := ma.NewMultiaddr(*host)
+	if err != nil {
+		log.WithField("err", err).Fatal("NewMultiaddr() failed")
+	}
+	p := addr.Protocols()
+	if len(p) < 2 {
+		log.WithField("addr", addr).Fatal("need to protocolls in host flag.")
```
Comment (jbenet) [inline cmd/pollEndpoint/main.go]: `"need two protocols in host flag (/ip/tcp)"`

File: cmd/pollEndpoint/main.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+				f["temporary"] = opErr.Temporary()
+				log.WithFields(f).Println("net.OpError")
+			}
+		} else { // unexpected error from http.Get()
+			f["err"] = err
+			log.WithFields(f).Error("unknown error")
+		}
+
+		time.Sleep(*timeout)
+		*tries--
```
Comment (jbenet) [inline cmd/pollEndpoint/main.go]: i'd break this up into smaller funcs?

File: cmd/pollEndpoint/main.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+		resp, err := http.Get(u.String())
+
+		if err == nil {
+			resp.Body.Close()
+
+			if resp.StatusCode == http.StatusOK {
+				f["took"] = time.Since(start)
+				log.WithFields(f).Println("status ok - endpoint reachable")
+				os.Exit(0)
```
Comment (jbenet) [inline cmd/pollEndpoint/main.go]: i'd try to return out to a simple func like:

```go
func main() { 
  if err := run(); err != nil {
    log.Println("failed.")
    os.Exit(1)
  }
}
```

File: cmd/pollEndpoint/main.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
+// pollEndpoint is a helper utility that waits for a http endpoint to be reachable and return with http.StatusOK
+package main
```
Comment (jbenet) [inline cmd/pollEndpoint/main.go]: maybe move this binary into `/thirdparty/pollEndpoint` ?

File: cmd/pollEndpoint/main.go (hunk 1 above)
Comment (cryptix) [inline cmd/pollEndpoint/main.go]: Done.

**Comment by jbenet** [comment]
this is my favorite PR in a while.

**Pull request merged**